spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerAutoConfiguration.java

```diff
@@ -0,0 +1,126 @@
+import static io.kubernetes.client.util.Namespaces.NAMESPACE_DEFAULT;
+
+/**
+ * @author wind57
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnDiscoveryEnabled
+@ConditionalOnKubernetesDiscoveryEnabled
+@ConditionalOnBlockingOrReactiveEnabled
+@ConditionalOnCloudPlatform(CloudPlatform.KUBERNETES)
+@AutoConfigureBefore({ SimpleDiscoveryClientAutoConfiguration.class, CommonsClientAutoConfiguration.class })
+@AutoConfigureAfter({ KubernetesClientAutoConfiguration.class, KubernetesDiscoveryPropertiesAutoConfiguration.class })
+public class KubernetesInformerAutoConfiguration {
+
+	private static final LogAccessor LOG = new LogAccessor(
+			LogFactory.getLog(KubernetesInformerAutoConfiguration.class));
+
+	@Bean
+	@ConditionalOnMissingBean
+	public SharedInformerFactory sharedInformerFactory(ApiClient client) {
+		return new SharedInformerFactory(client);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(value = V1Service.class, parameterizedContainer = SharedIndexInformer.class)
+	public SharedIndexInformer<V1Service> servicesSharedIndexInformer(SharedInformerFactory sharedInformerFactory,
+			ApiClient apiClient, KubernetesNamespaceProvider kubernetesNamespaceProvider,
+			KubernetesDiscoveryProperties discoveryProperties) {
+
+		GenericKubernetesApi<V1Service, V1ServiceList> servicesApi = new GenericKubernetesApi<>(V1Service.class,
+				V1ServiceList.class, "", "v1", "services", apiClient);
+
+		return sharedInformerFactory.sharedIndexInformerFor(servicesApi, V1Service.class, 0L,
+				namespace(discoveryProperties, kubernetesNamespaceProvider));
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(value = V1Endpoints.class, parameterizedContainer = SharedIndexInformer.class)
+	public SharedIndexInformer<V1Endpoints> endpointsSharedIndexInformer(SharedInformerFactory sharedInformerFactory,
+			ApiClient apiClient, KubernetesNamespaceProvider kubernetesNamespaceProvider,
+			KubernetesDiscoveryProperties discoveryProperties) {
+
+		GenericKubernetesApi<V1Endpoints, V1EndpointsList> servicesApi = new GenericKubernetesApi<>(V1Endpoints.class,
+				V1EndpointsList.class, "", "v1", "endpoints", apiClient);
+
+		return sharedInformerFactory.sharedIndexInformerFor(servicesApi, V1Endpoints.class, 0L,
+				namespace(discoveryProperties, kubernetesNamespaceProvider));
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(value = V1Service.class, parameterizedContainer = Lister.class)
+	public Lister<V1Service> servicesLister(SharedIndexInformer<V1Service> servicesSharedIndexInformer) {
+		return new Lister<>(servicesSharedIndexInformer.getIndexer());
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(value = V1Endpoints.class, parameterizedContainer = Lister.class)
+	public Lister<V1Endpoints> endpointsLister(SharedIndexInformer<V1Endpoints> endpointsSharedIndexInformer) {
+		return new Lister<>(endpointsSharedIndexInformer.getIndexer());
+	}
+
+	private String namespace(KubernetesDiscoveryProperties discoveryProperties,
```

Comment:
this logic was verbatim taken as it was before in the previous code. It lacks support for "selective namespaces", but I will add it in a different PR